include/zephyr/dt-bindings/pinctrl/hpmicro-pinctrl-common.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 hpmicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_HPMICRO_PINCTRL_COMMON_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_HPMICRO_PINCTRL_COMMON_H_
+
+/**
+ * @brief Pin numbers of gpio
+ *
+ */
+#define HPMICRO_PIN_NUM_SHIFT		0U
+#define HPMICRO_PIN_NUM_MASK		0x3FFU
+
+/**
+ * @brief io peripheral function
+ *
+ */
+#define HPMICRO_PIN_ANALOG_MASK		1U
+#define HPMICRO_PIN_ANALOG_SHIFT	10U
+#define HPMICRO_PIN_ALT_MASK		0x1FU
+#define HPMICRO_PIN_ALT_SHIFT		11U
+#define HPMICRO_PIN_IOC_MASK		0x03U
+#define HPMICRO_PIN_IOC_SHIFT		30U
```

Comment:
Masks can be defined with `BIT_MASK` macro and other helper macros. Please use them where appropriate.